Snowflake+transient (#9695)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/snowflake-configs.md

Snowflake supports the creation of [transient tables](https://docs.snowflake.net/manuals/user-guide/tables-temp-transient.html). Snowflake does not preserve a history for these tables, which can result in a measurable reduction of your Snowflake storage costs. Transient tables participate in time travel to a limited degree with a retention period of 1 day by default with no fail-safe period. Weigh these tradeoffs when deciding whether or not to configure your dbt models as `transient`. **By default, all Snowflake tables created by dbt are `transient`.**

:::note `transient` does not apply to seeds

> On Snowflake, the `transient` config applies to models, not seeds. If you set `+transient: true` or `+transient: false` on a seed, dbt still creates a _permanent_ table, and `dbt parse` might not warn you. This behavior applies to dbt Core and the dbt Fusion engine. For related discussion, refer to [dbt-snowflake#396](https://github.com/dbt-labs/dbt-snowflake/issues/396).

Author: ```suggestion
On Snowflake, the `transient` config applies to models, not seeds. If you set `+transient: true` or `+transient: false` on a seed, dbt still creates a _permanent_ table, and `dbt parse` might not warn you. This behavior applies to <Constant name="core" /> and the <Constant name="fusion_engine" />. For related discussion, refer to [dbt-snowflake#396](https://github.com/dbt-labs/dbt-snowflake/issues/396).
```

> :::

Reviewer: I think this is the wrong note.

We should just say this is a _model_ config.

(instead of saying it doesn't apply to seeds. i imagine it doesn't apply to snapshots too?)

Snowflake supports the creation of [transient tables](https://docs.snowflake.net/manuals/user-guide/tables-temp-transient.html). Snowflake does not preserve a history for these tables, which can result in a measurable reduction of your Snowflake storage costs. Transient tables participate in time travel to a limited degree with a retention period of 1 day by default with no fail-safe period. Weigh these tradeoffs when deciding whether or not to configure your dbt models as `transient`. **By default, all Snowflake tables created by dbt are `transient`.**

:::note `transient` is a model configuration

> 

Reviewer: Oh, if it applies to snapshots then it's a snapshot and model config. Have we confirmed this?

> On Snowflake, `transient` is a [model configuration](/reference/model-configs). It is also applied when a [snapshot](/docs/build/snapshots) creates its target table for the first time.

Reviewer: ```suggestion
On Snowflake, `transient` is a configuration for [models](/reference/model-configs) and [snapshots](/docs/build/snapshots).
```
